### PR TITLE
Update sendCross method

### PR DESCRIPTION
**Pull request by dnkolegov**
This PR fixes https://github.com/consensus-shipyard/ipc-solidity-actors/issues/90

### CODE REVIEW

File: src/Gateway.sol [hunk 1]
```diff
@@ -502,10 +503,13 @@ contract Gateway is IGateway, ReentrancyGuard, Voting {
...
 
-        // we disregard the "to" of the message. the caller is the one set as the "from" of the message.
-        crossMsg.message.to.subnetId = destination;
-        crossMsg.message.from.subnetId = _networkName;
-        crossMsg.message.from.rawAddress = msg.sender;
+        // we disregard the "to" of the message. the caller is the one set as the "from" of the message
+        if (!crossMsg.message.from.subnetId.equals(_networkName)) {
+            revert InvalidCrossMsgFromSubnetId();
+        }
+        if (crossMsg.message.from.rawAddress != msg.sender) {
```
Comment (adlrocha) [inline src/Gateway.sol:510]: I don't think I agree with this check, why are we doing this? Actually, I am starting to consider that we should not do anything to the from and simply add the cross-net information so we can have an arbitrary `from`, or force the `from` to be the actor triggering the message.

**Review by adlrocha — APPROVED** [review]
My only concern, is the `from` check, I don't know if we should remove it completely :shrug:

File: src/Gateway.sol (hunk 1 above)
Comment (dnkolegov) [reply src/Gateway.sol:510]: This line I derived from the original code - https://github.com/consensus-shipyard/ipc-solidity-actors/blob/main/src/Gateway.sol#L508.

Comment (adlrocha) [reply src/Gateway.sol:510]: did you maybe share the wrong link? This seems from the Solidity actors.

File: src/Gateway.sol [hunk 2]
```diff
@@ -487,25 +490,29 @@ contract Gateway is IGateway, ReentrancyGuard, Voting {
...
-        // we disregard the "to" of the message. the caller is the one set as the "from" of the message.
-        crossMsg.message.to.subnetId = destination;
-        crossMsg.message.from.subnetId = _networkName;
-        crossMsg.message.from.rawAddress = msg.sender;
+        // We disregard the "to" of the message that will be verified in the _commitCrossMessage().
+        // The caller is the one set as the "from" of the message
+        if (!crossMsg.message.from.subnetId.equals(_networkName)) {
+            revert InvalidCrossMsgFromSubnetId();
+        }
+        // The can be many semantics of the (rawAddress, msg.sender) pairs.
```
Comment (adlrocha) [inline src/Gateway.sol:508]:
```suggestion
        // There can be many semantics of the (rawAddress, msg.sender) pairs.
```

**Review by adlrocha — APPROVED** [review]
LGTM. I think we can merge.

**Review by adlrocha — APPROVED** [review]

**Pull request merged**